[MRG] Updated release notes for version 1.0

mrbean-bremen opened the pull request:
- adapted links in README
- see #523

<!--
Thanks for contributing a pull request! Please ensure you have taken a look at
the contribution guidelines: https://github.com/pydicom/pydicom/blob/master/CONTRIBUTING.md#contributing-pull-requests
-->
#### Reference Issue
<!-- Example: Fixes #1234 -->


#### What does this implement/fix? Explain your changes.
I checked all checkins since the last release and tried to compile a list of relevant changes.
Also updated links in README to point to GitHub Pages.

darcymason commented on doc/whatsnew/v1.0.0.rst:
```diff
@@ -7,43 +7,75 @@ changes.
+* optional Pillow and jpeg_ls support for reading some compressed pixel data
+  files
+* support for both decompressing a dataset inline, and return a decompressed
+  dataset
```
slight error in the language here, I think -- we don't return a decompressed dataset, but convert the dataset to decompressed in-place, no returned values.  The other option is for pixel_array to return decompressed pixel data without modifying the dataset otherwise.

darcymason replied:
Also, I'd like to see a link here (right where the new handling of compressed pixel data is announced) to a page with details/warnings - including @rhaxton's list in #542 of what works with each kind of compression, and what is giving uncertain results.  With that link, I'd suggest we strongly advise that people read that page before using these handlers.

mrbean-bremen replied:
> slight error in the language here

Ah yes, of course. I will just remove the part about returning the compressed data, that is covered by the prior 2 items.
> Also, I'd like to see a link here

Good idea - I will add something in the evening.

darcymason commented on doc/whatsnew/v1.0.0.rst:
```diff
@@ -7,43 +7,75 @@ changes.
-Other changes
-.............
+Infrastructure
+..............
```
I like this Infrastructure section.  Nice way to group things.

mrbean-bremen commented:
A question for @glemaitre - can I see the created documentation of a PR? As far as I understand, it is created after each PR check-in.

mrbean-bremen commented:
@darcymason - I added a very preliminary chapter for the image handler stuff - this has to be reworked by someone who knows the stuff better. Also, I haven't checked yet if the link works...

glemaitre commented:
‎From the top of the head, click on the circle ci icon it will redirect to the ci build. Add #artifacts in the url and then click in in the doc folder until you get index.htmlWhen I have a computer I will check if this working.

mrbean-bremen commented:
Ah, thanks a lot - this works! Nice!

darcymason commented:
I really like the data handlers chapter -- well explained, I don't see any need to correct anything. 
In fact, I'd like to see it linked in other places too.  In the side-bar it follows the Pydicom User Guide chapter, which has a 'Working with Pixel Data' section, where it should be prominent too.  Maybe it just should be put in the User Guide alongside that section?  At the very least it should be linked from there too.

Do we want to merge these documentation changes made so far, and then work on things like that in stages?  There are still many things to correct/update I see as I look through the ci 'artifact' build.

mrbean-bremen commented:
Thanks for the nice feedback! I moved the chapter into the user guide as suggested and made a few changes to match the user guide better.
I added a couple of usage line - maybe there shall go some example code, too - not sure if that is needed. 
Also I would like @rhaxton to have a look at the supported transfer syntaxes part to check if that is correct.

About merging - I generally like incremental work, but I leave that to you. Maybe you want to compile a list of the needed corrections to the user guide, so that someone can work on them? Or are these small things that you just want to fix yourself? I can look at them in the evening, too.
